showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -114,6 +154,109 @@ class ShowkaseProcessor @JvmOverloads constructor(
+
+        // Supported annotations from classpath
+        val supportedCustomPreview = environment.getTypeElementsFromPackage(CODEGEN_PACKAGE_NAME)
+            .flatMap { it.getEnclosedElements() }.mapNotNull {
+                when (val annotation =
+                    it.getAnnotation(ShowkaseMultiPreviewCodegenMetadata::class)) {
+                    null -> null
+                    else -> annotation.value
+                }
+            }
+        return supportedCustomPreview.mapIndexed { index: Int, customPreviewMetadata: ShowkaseMultiPreviewCodegenMetadata ->
+            roundEnvironment.getElementsAnnotatedWith(customPreviewMetadata.supportTypeQualifiedName)
+                .mapIndexed { elementIndex, xElement ->
+                    if (showkaseValidator.checkElementIsAnnotationClass(xElement)) {
+                        // Here we write to metadata to aggregate custom annotation data.
+                        // In this case, it would be a custom annotation that is annotated
+                        // with something custom. Eg. CombinedPreview.
+                        ShowkaseBrowserWriter(environment).writeCustomAnnotationElementToMetadata(
+                            xElement
+                        )
+                        return@mapIndexed null
+                    }
+                    showkaseValidator.validateComponentElement(
+                        xElement,
+                        customPreviewMetadata.supportTypeQualifiedName,
+                    )
+                    val commonMetadata = xElement.extractCommonMetadata(showkaseValidator)
+                    val previewParamMetadata = xElement.getPreviewParameterMetadata()
+                    ShowkaseMetadata.Component(
+                        element = xElement,
+                        elementName = xElement.name,
+                        packageName = commonMetadata.packageName,
+                        packageSimpleName = commonMetadata.moduleName,
+                        showkaseName = getShowkaseName(
+                            "${customPreviewMetadata.previewName}_$elementIndex",
+                            xElement.name
+                        ),
+                        insideObject = commonMetadata.showkaseFunctionType.insideObject(),
+                        previewParameterName = customPreviewMetadata.previewName,
+                        previewParameterProviderType = previewParamMetadata?.second,
+                        showkaseGroup = getShowkaseGroup(
+                            customPreviewMetadata.previewGroup,
+                            commonMetadata.enclosingClass
+                        ),
+                        showkaseKDoc = commonMetadata.kDoc,
+                        enclosingClassName = commonMetadata.enclosingClassName,
+                        componentIndex = elementIndex + index, // Need this to be unique because of filtering algorithm. TOOD: Find a better way to do this
```

Comment:
This I am unsure about. I kinda need this to be unique to remove the index from the name 🤔